other/upgrade_2-0_postgresql.sql

```diff
@@ -918,192 +918,67 @@ if (!isset($modSettings['attachment_thumb_png']))
-		if (!isset($core['theme_url']))
-			$core['theme_url'] = $GLOBALS['boardurl'] . '/Themes/core';
-		if (!isset($core['images_url']))
-			$core['images_url'] = $GLOBALS['boardurl'] . '/Themes/core/images';
-
-		// Get an available id_theme first...
-		$request = upgrade_query("
-			SELECT MAX(id_theme) + 1
-			FROM {$db_prefix}themes");
-		list ($id_core_theme) = $smcFunc['db_fetch_row']($request);
-		$smcFunc['db_free_result']($request);
-
-		// Insert the core theme into the tables.
-		$smcFunc['db_insert']('ignore',
-			'{db_prefix}themes',
-				array('id_member' => 'int', 'id_theme' => 'int', 'variable' => 'string-255', 'value' => 'string-255'),
-				array(
-					array(0, $id_core_theme, 'name', 'Core Theme'),
-					array(0, $id_core_theme, 'theme_url', $core['theme_url']),
-					array(0, $id_core_theme, 'images_url', $core['images_url']),
-					array(0, $id_core_theme, 'theme_dir', $core['theme_dir'])
-				),
-				array()
-		);
+		$row = smf_mysql_fetch_row($theme_request);
+		$id_theme = $row[0];
+		smf_mysql_free_result($theme_request);
+		unset($row);
 
-		// Update the name of the default theme in the database.
-		upgrade_query("
-			UPDATE {$db_prefix}themes
-			SET value = 'SMF Default Theme - Curve'
-			WHERE id_theme = 1
-				AND variable = 'name'");
-
-		$newSettings = array();
-		// Now that we have the old theme details - switch anyone who used the default to it (Make sense?!)
-		if (!empty($modSettings['theme_default']) && $modSettings['theme_default'] == 1)
-			$newSettings[] = "('theme_default', $id_core_theme)";
-		// Did guests use to use the default?
-		if (!empty($modSettings['theme_guests']) && $modSettings['theme_guests'] == 1)
-			$newSettings[] = "('theme_guests', $id_core_theme)";
-
-		// If known themes aren't set, let's just pick all themes available.
-		if (empty($modSettings['knownThemes']))
-		{
-			$request = upgrade_query("
-				SELECT DISTINCT id_theme
-				FROM {$db_prefix}themes");
-			$themes = array();
-			while ($row = $smcFunc['db_fetch_assoc']($request))
-				$themes[] = $row['id_theme'];
-			$modSettings['knownThemes'] = implode(',', $themes);
-			upgrade_query("
-				UPDATE {$db_prefix}settings
-				SET value = '$modSettings[knownThemes]'
-				WHERE variable = 'knownThemes'");
-		}
+		$known_themes = explode(',', $modSettings['knownThemes']);
 
-		// Known themes.
-		$allThemes = explode(',', $modSettings['knownThemes']);
-		$allThemes[] = $id_core_theme;
-		$newSettings[] = "('knownThemes', '" . implode(',', $allThemes) . "')";
+		// Remove this value...
+		$known_themes = array_diff($known_themes, array($id_theme));
 
-		// Since we want to do a replace, just delete the old settings and re-insert them
+		// Change back to a string...
+		$known_themes = implode(',', $known_themes);
+
+		// Update the database
 		upgrade_query("
-			DELETE FROM {$db_prefix}settings
-			WHERE variable IN ('theme_default', 'theme_guests', 'knownThemes')");
+			REPLACE INTO {$db_prefix}settings (variable, value)
```

Comment:
Use the replace methode of $smcfunc['db_insert'],
replace got a different syntax in pg.

Reply:
smcFunc replace used